.github/workflows/docker_publish.yml

```diff
@@ -0,0 +1,116 @@
+on:
+  push:
+    branches: ['gh_actions_and_ghcr_for_docker']  # todo replace with main once we know this is working
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+
+    env:
+      hdf5_versions: 1.10.4
+      hdf5_build_dir: hdf5_build_dir
+
+    strategy:
+      matrix:
+       ubuntu_versions : [
+          16.04,
+          18.04,
+          ]
+       compiler : [
+          gcc,
+          clang,
+          ]
+       hdf5_versions : [
+         1.10.4,
+       ]
+       moab_versions : [
+         9c96d17,
+         develop,
+         master,
+       ]
+
+
+
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Verify Changed files
```

Comment:
I think this action is more complex than necessary.... and I read that it's for detecting changes during the action/workflow

It should be possible to add some version of `paths:` to the top level job control in the first few lines.